File: Producto/routes-service/src/main/java/com/tagok/routes_service/dto/RouteSegment.java
package com.tagok.routes_service.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class RouteSegment 
{
    private Integer seq;
    private Long edgeId;
    private Long node;

    private Double cost;
    private Double aggCost;

    private String name;

    private String geometry;
}

File: Producto/tag-ok-app/app/src/main/java/com/tagok/app/data/dto/RouteSegment.kt
package com.tagok.app.data.dto

import kotlinx.serialization.Serializable

@Serializable
data class RouteSegment(
    val seq: Int = 0,
    val edgeId: Long = 0,
    val node: Long = 0,
    val cost: Double = 0.0,
    val aggCost: Double = 0.0,
    val name: String? = null,
    val geometry: String = "")   // JSON string: {"type":"LineString","coordinates":[[lon,lat],...]}